dist/init/linux-upstart/caddy.conf

```diff
@@ -14,12 +14,12 @@ respawn limit 10 5
 reload signal SIGUSR1
 
 # Let's Encrypt certificates will be written to this directory.
-env CADDYPATH=/etc/caddy
+env CADDYPATH=/etc/ssl/caddy
```

Comment:
As far as content is concerned, I see that `/etc/caddy` contains only `Caddyfile` so I do not expect conflicts. The main problem is permissions:
* systemd has specific [instructions](https://github.com/mholt/caddy/blob/6cabc9bfe3dc5f06c49aa8fb58ac19a09a4f42e9/dist/init/linux-systemd/README.md#instructions) and [code](https://github.com/mholt/caddy/blob/6cabc9bfe3dc5f06c49aa8fb58ac19a09a4f42e9/dist/init/linux-systemd/caddy.service#L34-L38) for distinct permissions of the two folders;
* The other init systems have notes on permissions, but not as specific as systemd.

Having the two folders distinct rather than one-subfolder-than-the-other shall make permission setting simpler.